oauth2client/service_account.py

```diff
@@ -426,6 +432,32 @@ def create_scoped(self, scopes):
         result._private_key_pkcs12 = self._private_key_pkcs12
         result._private_key_password = self._private_key_password
         return result
+ 
+    def create_with_claims(self, claims):
+        """Create credentials that specify additional claims.
+
+        Args:
+            claims: dict, key-value pairs for claims.
+
+        Returns:
+            ServiceAccountCredentials, a copy of the current service account
+            credentials with updated claims to use when obtaining access tokens.
+        """
+        new_kwargs = dict(self._kwargs)
+        new_kwargs.update(claims)
+        result = self.__class__(self._service_account_email,
```

Comment:
`self.__class__` is a little surprising here - I guess I thought this would have thought this would always return instances of one specific class. Am I just missing the larger picture?